developer/reference/articles/tooling/blade-cli/08-extending-blade-cli/05-creating-a-blade-profile.markdown

```diff
@@ -0,0 +1,113 @@
+
+3.  To override a default command, create new command and arguments classes
+    similar to what was done
+    [here](/docs/7-2/reference/-/knowledge_base/r/creating-custom-commands-for-blade-cli).
+    Instead of extending the
+    [`BaseCommand`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseCommand.java)
+    and
+    [`BaseArgs`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseArgs.java),
+    classes, extend the command/arguments classes defined for the command you
+    intend to override. Make sure to also set the `@Parameters` annotation's
+    `commandNames` argument to the command to override.
+
+    For example, if you intend to override the default `deploy` command, your
+    arguments class declaration would look like this:
+
+    ```java
+    @Parameters(commandDescription = "Overridden Deploy Command", commandNames = "deploy")
+    public class OverriddenArgs extends DeployArgs {
+
+    }
+    ```
+
+    You can search for the default command/arguments classes
+    [here](https://github.com/liferay/liferay-blade-cli/tree/master/cli/src/main/java/com/liferay/blade/cli/command).
+
+4.  To associate a command to your new profile, set the
+    [`BladeProfile`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BladeProfile.java)
+    annotation in your command class:
+
+    ```java
+    @BladeProfile("myprofile")
+    public class NewCommand extends BaseCommand<NewArgs> {
+
+    }
+    ```
+
+    The annotation's parameter should specify the profile you want to associate
+    the command with (e.g., `myprofile`).
+
+Excellent! You've created a new Blade profile and learned how to add new
+commands or override default commands by leveraging the profile. You can
+reference the
+[sample profile project](https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-profile)
+to examine a new command and overridden command's setup in a custom profile.
+
+Next, you'll learn how to set your new profile for use in a Liferay Workspace.
+
+## Setting a Profile
+
+To set your new Blade profile in a Liferay Workspace, open the
+`${workspaceDir}/.blade.properties` file and set the `profile.name` property to
+your profile name:
+
+```properties
+profile.name=myprofile
+```
+
+This specifies which Blade profile is active and uses the commands with that
+defined profile. The default setting is `gradle`. You can also set this property
+to `maven` out-of-the-box, which is applied for Maven-based workspaces.
+
+You can specify the Blade profile for a workspace when initializing it too. This
+is done by passing the profile name as an argument when creating the workspace:
+
+```bash
+blade init -p <profile-name> <workspace-name>
+```
+
+For example, if you execute the following command:
+
+```bash
+blade init -p myprofile my-new-custom-workspace
+```
+
```

Comment:
We may want to mention that the user can use the -P option for any command, to specify the profile to be used, it's not specific to `init`